mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/master/resourcecluster/ResourceCluster.java

```diff
@@ -97,6 +97,8 @@ default CompletableFuture<List<TaskExecutorID>> getUnregisteredTaskExecutors() {
 
     CompletableFuture<Ack> addNewJobArtifactsToCache(ClusterID clusterID, List<ArtifactID> artifacts);
 
+    CompletableFuture<Ack> markExecutorTaskCancelled(WorkerId workerId);
+
```

Comment:
Why do you need an explicit method here? Can we somehow listen in on the event that's sent to the TaskExecutorGateway to know that the TE has been asked to shutdown.

Reply:
It looks like this method is only being called during scheduling failures. It should also be called whenever a worker is sending heartbeats that the JobActor / JobClusterActor is not acknowledging.